consensus/istanbul/core/commit.go

```diff
@@ -107,11 +107,16 @@ func (c *core) handleCommit(msg *message, src istanbul.Validator) error {
 			logger.Warn("received commit of the hash locked proposal and change state to prepared", "msgType", msgCommit)
 			c.setState(StatePrepared)
 			c.sendCommit()
-		} else if c.current.GetPrepareOrCommitSize() > 2*c.valSet.F() {
+		} else if c.current.GetPrepareOrCommitSize() > 2*c.valSet.F() && c.valSet.Size() > ExceptionalValidatorsNumber {
 			logger.Info("received more than 2f agreements and change state to prepared", "msgType", msgCommit)
 			c.current.LockHash()
 			c.setState(StatePrepared)
 			c.sendCommit()
+		} else if uint64(c.current.GetPrepareOrCommitSize()) == c.valSet.Size() && c.valSet.Size() <= ExceptionalValidatorsNumber {
+			logger.Info("received all of agreements and change state to prepared", "msgType", msgCommit)
```

Comment:
I suggest showing `c.valSet.Size()` when it is too small (equal or below than `ExceptionalValidatorsNumber`)

`logger.Info("due to small valset size, received all of agreements and change state to prepared", "msgType", msgCommit, "valset", c.valSet.Size() )`

Reply:
Thanks for your review.
Fixed in [af09bd6](https://github.com/klaytn/klaytn/pull/1462/commits/af09bd6b3bd296e33b1e125dcd4b96665c5c865c).